carla_ackermann_control_wrapper/src/carla_ackermann_control_wrapper.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "carla_ackermann_control_wrapper.h"
+
+namespace carla_ackermann_control_wrapper
+{
+CarlaAckermannControlWrapper::CarlaAckermannControlWrapper(){}
+
+void CarlaAckermannControlWrapper::init()
+{
+    ROS_INFO("Initalizing carla ackermann control wrapper node...");
+
+    // Load parameters
+    wheel_base_ = pnh_.param<double>("vehicle_wheel_base", 2.79);
+    max_steer_angle_ = pnh_.param<double>("max_steer_angle", 29.2);
+    max_speed_ = pnh_.param<double>("max_speed", 50);
+    vehicle_mass_ = pnh_.param<double>("vehicle_weight", 1500);
+    max_accel_ = pnh_.param<double>("acceleration_limit", 3.0);
+    max_decel_ = pnh_.param<double>("deceleration_limit", 6.0);
+
+    // Set driver type
+    driver_status_.controller = true;
+}
+
+// Publish robotic status
+void CarlaAckermannControlWrapper::robot_status_cb(const cav_msgs::CarlaEnabledConstPtr& msg)
+{
+    robotic_status_.robot_active = msg->carla_enabled;
+    robot_status_pub_.publish(robotic_status_);
+    update_controller_health_status();
+}
+
+void CarlaAckermannControlWrapper::update_controller_health_status()
+{
+    if (robotic_status_.robot_active)
```

Comment:
I see Guidance node subscribes robot_status, so I publish the robot_status in function robot_status_cb. When the message CarleEnabled is ture, it will set the robot_status to true and then set the controller status to operational. Then Carla knows Carma is ready and they start synchronizing data.

Reply:
I think the controller needs to be operational to complete system startup/operation, though. If the controller isn't flagged as operational, the system health monitor will cause a shutdown. The operational flag is meant to say that the controller driver is running and successfully talking to the controller hardware as a pre-requisite for being able to control the vehicle.

Reply:
Yes, the controller is setup in the init(). Then when the robotic_status is active, the controller will be updated to operational.

Reply:
Modified to set the controller to operational in init()